src/text-fragment-utils.js

```diff
@@ -598,8 +598,16 @@ function* getElementsIn(root, filter) {
  */
 const findTextInRange = (query, range) => {
   const textNodeLists = getAllTextNodes(range.commonAncestorContainer, range);
+  let segmenter = null;
+  if (Intl.Segmenter) {
+    let lang = document.documentElement.lang;
```

Comment:
How reliable is this? What if some parts of the text are not in the same locale as the overall document? What is the impact of using a segmenter that's in the wrong language?

Reply:
There's not a nice, easy answer to this question, because the spec leaves it up to the implementers how to actually segment the words (and only "recommends" that the locale be taken into account). In my testing segmenting various strings, I didn't see a difference in behavior regardless of the locale passed (be it "en", "jp", or even "whatever") on either WebKit or Chrome (Blink). 

So my takeaway is, we have to provide *something* here and there's a chance that browsers will start using it in the future, but it doesn't seem to make a difference right now. I think the language provided by the page author is as reliable as any other guess we could make.